src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1392,21 +1389,22 @@ private bool HasIdentityConversionInternal(TypeSymbol type1, TypeSymbol type2)
             return HasIdentityConversionInternal(type1, type2, IncludeNullability);
         }
 
-        internal static bool HasTopLevelNullabilityIdentityConversion(TypeSymbolWithAnnotations source, TypeSymbolWithAnnotations destination)
+        internal bool HasTopLevelNullabilityIdentityConversion(TypeSymbolWithAnnotations source, TypeSymbolWithAnnotations destination)
         {
-            bool? sourceIsNullable = source.IsNullable;
-            bool? destinationIsNullable = destination.IsNullable;
-            return sourceIsNullable == null || destinationIsNullable == null || sourceIsNullable == destinationIsNullable;
-        }
-
-        internal static bool HasTopLevelNullabilityImplicitConversion(TypeSymbolWithAnnotations source, TypeSymbolWithAnnotations destination)
-        {
-            return HasTopLevelNullabilityImplicitConversion(source.IsNullable, destination.IsNullable);
+            if (!IncludeNullability)
+            {
+                return true;
+            }
+            return source.IsNullable == destination.IsNullable;
         }
 
-        internal static bool HasTopLevelNullabilityImplicitConversion(bool? sourceIsNullable, bool? destinationIsNullable)
+        internal bool HasTopLevelNullabilityImplicitConversion(TypeSymbolWithAnnotations source, TypeSymbolWithAnnotations destination)
         {
-            return sourceIsNullable != true || destinationIsNullable != false;
+            if (!IncludeNullability)
+            {
+                return true;
+            }
+            return source.IsNullable != true || destination.IsNullable != false;
```

Comment:
Consider inverting this conditional to explicitly spell out the one case that does not have an implicit conversion, rather than spelling out all the cases that do.

Reply:
The alternative, `!(source.IsNullable == true && destination.IsNullable == false)`, seems equally confusing to me.

---
In reply to: [223834417](https://github.com/dotnet/roslyn/pull/29727#discussion_r223834417) [](ancestors = 223834417)